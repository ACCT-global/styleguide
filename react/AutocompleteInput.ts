export { default } from './components/Autocomplete'
