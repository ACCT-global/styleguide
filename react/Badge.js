import Tag from './components/Tag/index'
import deprecated from './modules/deprecated'

export default deprecated(
  {
    useNewComponent: {
      Badge: 'Tag',
    },
  },
  Tag
)
