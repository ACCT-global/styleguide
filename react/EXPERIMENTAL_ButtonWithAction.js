import EXPERIMENTAL_ButtonWithAction from './components/EXPERIMENTAL_ButtonWithAction/index'

export default EXPERIMENTAL_ButtonWithAction
