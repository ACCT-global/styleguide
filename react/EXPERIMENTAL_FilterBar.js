import EXPERIMENTAL_FilterBar from './components/EXPERIMENTAL_FilterBar/index'

export default EXPERIMENTAL_FilterBar
