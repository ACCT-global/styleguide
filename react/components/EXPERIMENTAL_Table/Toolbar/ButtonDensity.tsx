import React, { FC } from 'react'

import IconDensity from '../../icon/Density/index.js'
import constants from '../constants'
import useTableContext from '../hooks/useTableContext'
import Menu from './Menu/index'

const BOX_HEIGHT =
  constants.DENSITY_OPTIONS.length * constants.FIELDS_BOX_ITEM_HEIGHT

type DensityProps = {
  label: string
  lowOptionLabel: string
  mediumOptionLabel: string
  highOptionLabel: string
  handleCallback: Function
  alignMenu: 'right' | 'left'
  disabled: boolean
}

const ButtonDensity: FC<DensityProps> = ({
  label,
  handleCallback,
  disabled,
  alignMenu,
  ...options
}) => {
  const { selectedDensity, setSelectedDensity } = useTableContext()
  return (
    <Menu
      button={{
        id: 'toggleDensity',
        title: label,
        icon: <IconDensity size={constants.MEDIUM_ICON_SIZE} />,
        disabled,
      }}
      box={{ height: BOX_HEIGHT, alignMenu }}>
      {constants.DENSITY_OPTIONS.map((key: Density, index) => {
        const isKeySelected = selectedDensity === key
        return (
          <Menu.Item
            key={index}
            isSelected={isKeySelected}
            handleCallback={() => {
              setSelectedDensity(key)
              handleCallback && handleCallback(key)
            }}
            closeMenuOnClick>
            {options[`${key}OptionLabel`]}
          </Menu.Item>
        )
      })}
    </Menu>
  )
}

export default ButtonDensity
