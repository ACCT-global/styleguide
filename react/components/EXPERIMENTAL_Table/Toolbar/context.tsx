import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from '../types'

const ToolbarContext = createContext<E2ETestable>(null)

export const ToolbarProvider: FC<E2ETestable> = ({ children, testId }) => {
  return (
    <ToolbarContext.Provider value={{ testId }}>
      {children}
    </ToolbarContext.Provider>
  )
}

export function useToolbarContext() {
  const context = useContext(ToolbarContext)
  if (!context) {
    throw new Error('Do not use Toolbar composites outside of context')
  }
  return context
}
