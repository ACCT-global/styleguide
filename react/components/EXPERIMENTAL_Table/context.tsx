import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from './types'
import useTableMeasures from './hooks/useTableMeasures'

type Measures = ReturnType<typeof useTableMeasures>

const TestingContext = createContext<E2ETestable>(null)
const MeasuresContext = createContext<Measures>(null)

type TableType = E2ETestable & {
  measures: Measures
}

export const TableProvider: FC<TableType> = ({
  children,
  testId,
  measures,
}) => {
  return (
    <TestingContext.Provider value={{ testId }}>
      <MeasuresContext.Provider value={measures}>
        {children}
      </MeasuresContext.Provider>
    </TestingContext.Provider>
  )
}

export function useTestingContext() {
  const context = useContext(TestingContext)
  if (!context) {
    throw new Error(
      'Do not use E2ETestable components outside of the TestingContext'
    )
  }
  return context
}

export function useMeasuresContext() {
  const context = useContext(MeasuresContext)
  if (!context) {
    throw new Error(
      'Do not use measurable components outside of the MeasuresContext'
    )
  }
  return context
}
