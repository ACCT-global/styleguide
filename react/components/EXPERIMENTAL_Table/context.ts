import { createContext } from 'react'

const Context = createContext<TableState>(null)
const { Provider } = Context

export default Context
export { Provider as TableProvider }
