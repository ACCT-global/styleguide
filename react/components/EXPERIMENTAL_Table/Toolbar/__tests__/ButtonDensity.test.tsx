import React from 'react'
import { renderHook } from '@testing-library/react-hooks'
import { render, fireEvent, screen } from '@testing-library/react'

import { ButtonGroupProvider } from '../context'
import useTableMeasures from '../../hooks/useTableMeasures'
import { MeasuresProvider } from '../../context/measures'
import ButtonDensity from '../ButtonDensity'

describe('Table V2 @ Toolbar/ButtonDensity spec', () => {
  const basicProps = {
    label: 'Line density',
    compactLabel: 'Compact',
    regularLabel: 'Regular',
    comfortableLabel: 'Comfortable',
  }

  const { result } = renderHook(() =>
    useTableMeasures({
      size: 5,
    })
  )

  it('matches snapshot', () => {
    const button = render(
      <MeasuresProvider measures={result.current}>
        <ButtonGroupProvider testId="button-group">
          <ButtonDensity {...basicProps} />
        </ButtonGroupProvider>
      </MeasuresProvider>
    )

    expect(button.asFragment()).toMatchSnapshot()
  })
  it('matches snapshot after click', () => {
    const button = render(
      <MeasuresProvider measures={result.current}>
        <ButtonGroupProvider testId="button-group">
          <ButtonDensity {...basicProps} />
        </ButtonGroupProvider>
      </MeasuresProvider>
    )

    fireEvent.click(screen.getByTestId('button-group__density'))
    expect(button.asFragment()).toMatchSnapshot()
  })
})
