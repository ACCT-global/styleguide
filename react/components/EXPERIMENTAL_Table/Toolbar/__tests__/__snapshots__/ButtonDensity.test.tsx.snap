// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table V2 @ Toolbar/ButtonDensity spec matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="button-group__density__container"
    id="vtex-table-v2__toolbar__button-density__container"
    title="Line density"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small icon-button dib bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="button-group__density"
      style="font-size: 0px;"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph3 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="c-on-base"
          style="margin-top: 1.5px;"
        >
          <svg
            class="vtex__icon-density  "
            fill="currentColor"
            height="13"
            viewBox="0 0 14 13"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <rect
              fill="currentColor"
              height="1.85714"
              rx="0.928571"
              width="14"
              y="11.1426"
            />
            <rect
              fill="currentColor"
              height="1.85714"
              rx="0.928571"
              width="14"
              y="7.42871"
            />
            <rect
              fill="currentColor"
              height="5.57143"
              rx="1"
              width="14"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Table V2 @ Toolbar/ButtonDensity spec matches snapshot after click 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="button-group__density__container"
    id="vtex-table-v2__toolbar__button-density__container"
    title="Line density"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small icon-button dib bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="button-group__density"
      style="font-size: 0px;"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph3 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="c-on-base"
          style="margin-top: 1.5px;"
        >
          <svg
            class="vtex__icon-density  "
            fill="currentColor"
            height="13"
            viewBox="0 0 14 13"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <rect
              fill="currentColor"
              height="1.85714"
              rx="0.928571"
              width="14"
              y="11.1426"
            />
            <rect
              fill="currentColor"
              height="1.85714"
              rx="0.928571"
              width="14"
              y="7.42871"
            />
            <rect
              fill="currentColor"
              height="5.57143"
              rx="1"
              width="14"
            />
          </svg>
        </span>
      </div>
    </button>
    <div
      class="absolute z-999 shadow-4 b--muted-4 br2 ba left-0 mt2 mh2"
      data-testid="button-group__density__box"
    >
      <div
        class="w-100 b2 br2 bg-base"
      >
        <div
          class="overflow-auto"
          data-testid="button-group__density__box__items"
          style="height: 108px;"
        >
          <div
            class="flex justify-between ph6 pv3 pointer hover-bg-muted-5 bl bw1 b--transparent"
          >
            <span
              class="w-100 flex justify-between"
            >
              Compact
            </span>
          </div>
          <div
            class="flex justify-between ph6 pv3 pointer hover-bg-muted-5 bl bw1 b--emphasis"
          >
            <span
              class="w-100 flex justify-between fw5"
            >
              Regular
            </span>
          </div>
          <div
            class="flex justify-between ph6 pv3 pointer hover-bg-muted-5 bl bw1 b--transparent"
          >
            <span
              class="w-100 flex justify-between"
            >
              Comfortable
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
